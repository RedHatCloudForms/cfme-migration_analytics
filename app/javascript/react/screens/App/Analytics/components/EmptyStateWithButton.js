import React from 'react';
import PropTypes from 'prop-types';
import { Button, EmptyState, noop } from 'patternfly-react';

const EmptyStateWithButton = ({ iconType, iconName, title, message, buttonText, onClick, href }) => (
  <EmptyState>
    <EmptyState.Icon type={iconType} name={iconName} />
    <EmptyState.Title>{title}</EmptyState.Title>
    <EmptyState.Info>{message}</EmptyState.Info>
    <EmptyState.Action>
      <Button bsStyle="primary" bsSize="large" onClick={onClick} href={href}>
        {buttonText}
      </Button>
    </EmptyState.Action>
  </EmptyState>
);

EmptyStateWithButton.propTypes = {
  iconType: PropTypes.string,
  iconName: PropTypes.string,
  title: PropTypes.string.isRequired,
  message: PropTypes.node.isRequired,
  buttonText: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  href: PropTypes.string
};

EmptyStateWithButton.defaultProps = {
  iconType: 'pf',
  iconName: 'add-circle-o',
  onClick: noop,
  href: null
};

export default EmptyStateWithButton;
