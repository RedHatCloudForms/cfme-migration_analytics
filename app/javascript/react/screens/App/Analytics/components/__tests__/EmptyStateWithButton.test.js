import React from 'react';
import { shallow } from 'enzyme';
import EmptyStateWithButton from '../EmptyStateWithButton';

describe('empty state with button', () => {
  const getBaseProps = () => ({
    title: 'Title Goes Here',
    message: 'Message goes here',
    buttonText: 'Click me',
    onClick: jest.fn()
  });

  test('renders correctly', () => {
    const component = shallow(<EmptyStateWithButton {...getBaseProps()} />);
    expect(component).toMatchSnapshot();
  });

  test('calls onClick prop when primary button is clicked', () => {
    const props = getBaseProps();
    const component = shallow(<EmptyStateWithButton {...props} />);
    component.find('Button[bsStyle="primary"]').simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
});
