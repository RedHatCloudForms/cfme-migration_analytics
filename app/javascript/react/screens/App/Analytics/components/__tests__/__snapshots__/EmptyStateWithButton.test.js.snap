// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty state with button renders correctly 1`] = `
<EmptyState
  className=""
>
  <EmptyStateIcon
    className=""
    name="add-circle-o"
    type="pf"
  />
  <EmptyStateTitle
    className=""
  >
    Title Goes Here
  </EmptyStateTitle>
  <EmptyStateInfo
    className=""
  >
    Message goes here
  </EmptyStateInfo>
  <EmptyStateAction
    className=""
    secondary={false}
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="large"
      bsStyle="primary"
      disabled={false}
      href={null}
      onClick={[MockFunction]}
    >
      Click me
    </Button>
  </EmptyStateAction>
</EmptyState>
`;
